src/StripStylesTransformer.ts

```diff
@@ -0,0 +1,150 @@
+/*
+ * Code is inspired by
+ * https://github.com/kulshekhar/ts-jest/blob/25e1c63dd3797793b0f46fa52fdee580b46f66ae/src/transformers/hoist-jest.ts
+ *
+ * 
+ * IMPLEMENTATION DETAILS:
+ * This transformer handles one concern: removing styles.
+ *
+ * The property 'styles' inside a @Component(...) Decorator argument will
+ * be modified, even if they are not used in the context of an
+ * Angular Component.
+ * 
+ * This is the required AST to trigger the transformation:
+ * 
+ * ClassDeclaration
+ *   Decorator
+ *     CallExpression
+ *       ObjectLiteralExpression
+ *         PropertyAssignment
+ *           Identifier
+ *           StringLiteral
+ */
+
+// only import types, for the rest use injected `ConfigSet.compilerModule`
+import {
+  Node,
+  SourceFile,
+  TransformationContext,
+  Transformer,
+  Visitor,
+  Identifier,
+  ClassDeclaration,
+  PropertyAssignment
+} from 'typescript';
+import { ConfigSet } from './TransformUtils';
+
+/** Angular component decorator Styles property name */
+const STYLES = 'styles';
+/** Angular component decorator name */
+const COMPONENT = 'Component';
+/** All props to be transformed inside a decorator */
+const TRANSFORM_IN_DECORATOR_PROPS = [STYLES];
+
+/**
+ * Transformer ID
+ * @internal
+ */
+export const name = 'angular-component-strip-styles';
+
+// increment this each time the code is modified
+/**
+ * Transformer Version
+ * @internal
+ */
+export const version = 1;
+
+/**
+ * The factory of hoisting transformer factory
+ * @internal
+ */
+export function factory(cs: ConfigSet) {
+  /**
+   * Our compiler (typescript, or a module with typescript-like interface)
+   */
+  const ts = cs.compilerModule;
+
+  /**
+   * Traverses the AST down inside a decorator to a styles assignment
+   * and returns a boolean indicating if it should be transformed.
+   */
+  function isInDecoratorPropertyAssignmentToTransform(node: Node): node is ClassDeclaration {
+    return getInDecoratorPropertyAssignmentsToTransform(node).length > 0;
+  }
+
+  /**
+   * Traverses the AST down inside a decorator to a styles assignment
+   * returns it in an array.
+   */
+  function getInDecoratorPropertyAssignmentsToTransform(node: Node) {
+    if (!ts.isClassDeclaration(node) || !node.decorators) {
+      return [];
+    }
+
+    return node.decorators
+      .map(dec => dec.expression)
+      .filter(ts.isCallExpression)
+      .filter(callExpr => ts.isIdentifier(callExpr.expression) && callExpr.expression.getText() === COMPONENT)
+      .reduce((acc, nxtCallExpr) => acc.concat(
+        ...nxtCallExpr.arguments
+          .filter(ts.isObjectLiteralExpression)
+          .reduce((acc, nxtArg) => acc.concat(
+            ...nxtArg.properties
+              .filter(ts.isPropertyAssignment)
+              .filter(propAss => ts.isIdentifier(propAss.name))
+              .filter(propAss =>
```

Comment:
Can we combine these 2 filter conditions ?

Reply:
Combined the latter ones.